core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/util/HttpChecksumUtils.java

```diff
@@ -23,21 +25,18 @@
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
 import java.util.Optional;
-import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
 import software.amazon.awssdk.http.ContentStreamProvider;
 import software.amazon.awssdk.http.SdkHttpRequest;
-import software.amazon.awssdk.http.auth.internal.checksums.ChecksumAlgorithm;
-import software.amazon.awssdk.http.auth.internal.checksums.ChecksumSpecs;
-import software.amazon.awssdk.http.auth.internal.checksums.SdkChecksum;
-import software.amazon.awssdk.http.auth.internal.io.SdkDigestInputStream;
+import software.amazon.awssdk.http.auth.aws.checksum.SdkChecksum;
+import software.amazon.awssdk.http.auth.aws.internal.checksums.ChecksumSpecs;
+import software.amazon.awssdk.http.auth.aws.internal.io.SdkDigestInputStream;
 import software.amazon.awssdk.utils.BinaryUtils;
 import software.amazon.awssdk.utils.StringUtils;
 
-@SdkInternalApi
+@SdkProtectedApi
```

Comment:
Any way to avoid this needing to be protected? Or can we reduce the number of things it contains to strictly what is needed to be protected?

Reply:
I guess we can make things that aren't used cross-module package-private?